### ISSUE TITLE
Script combing EDEN_VAE and EDEN_DiT to EDEN

### ISSUE THREAD

**Issue by ajex1988** [bbldCVer/EDEN#17]
I noticed the authors released the pretrained vae and DiT, as well as the merged eden. But I can not find the script which takes vae and DiT as input and output eden. I tried to implement one but the results looks patchy. Any suggestion?

**Comment by bbldCVer** [comment]
Hi, sorry for not providing clear instructions in the README. We've discussed how to obtain eden.pt in https://github.com/bbldCVer/EDEN/issues/4 — hope that helps!

**Comment by ajex1988** [comment]
> Hi, sorry for not providing clear instructions in the README. We've discussed how to obtain eden.pt in [#4](https://github.com/bbldCVer/EDEN/issues/4) — hope that helps!

Thank you for reply.
I manually checked the weights of the pretrained vae, pretrained DiT and the final eden.pt. It seems that eden's weights are not coming from corresponding weights in vae and DiT. So I guess for the provided 3 checkpoints, eden.pt != combine(vae's decoder, DiT). Am I missing something?

**Comment by bbldCVer** [comment]
> > Hi, sorry for not providing clear instructions in the README. We've discussed how to obtain eden.pt in [#4](https://github.com/bbldCVer/EDEN/issues/4) — hope that helps!
> 
> Thank you for reply. I manually checked the weights of the pretrained vae, pretrained DiT and the final eden.pt. It seems that eden's weights are not coming from corresponding weights in vae and DiT. So I guess for the provided 3 checkpoints, eden.pt != combine(vae's decoder, DiT). Am I missing something?

Hi, thanks for bringing up this interesting observation!

After checking our previous code, we found that after **EDEN** was merged, we conducted several thousand steps of end-to-end fine-tuning to further enhance **EDEN**'s capabilities. Therefore, the final weights of `eden.pt` are different from the original `vae.pt` and `dit.pt`.

**Comment by ajex1988** [comment]
Thanks a lot, that make sense.
I also noticed that in the train_dit.yaml,   DAVIS has height: 256 and width: 448. In the eval_eden.yaml, DAVIS has height: 480 and width: 854. Any special consideration?

**Comment by bbldCVer** [comment]
> Thanks a lot, that make sense. I also noticed that in the train_dit.yaml, DAVIS has height: 256 and width: 448. In the eval_eden.yaml, DAVIS has height: 480 and width: 854. Any special consideration?

Good catch! This is intentional and relates to our training pipeline:

* **Pre-training Phase:** During the initial pre-training (low-resolution stage), we used a base resolution of **256x448**.
* **Model Sensitivity:** Since **EDEN** is a pure Transformer architecture, it is highly sensitive to input resolutions and positional embeddings.
* **Validation Consistency:** To maintain consistency with the pre-training data before moving into the multi-resolution fine-tuning stage, we perform validation using the 256x448 config.

The higher resolution you see in `eval_eden.yaml` (480x854) is intended for use after the model has undergone the multi-resolution fine-tuning phase.

Hope this clears things up!

**Comment by ajex1988** [comment]
> > > Hi, sorry for not providing clear instructions in the README. We've discussed how to obtain eden.pt in [#4](https://github.com/bbldCVer/EDEN/issues/4) — hope that helps!
> > 
> > 
> > Thank you for reply. I manually checked the weights of the pretrained vae, pretrained DiT and the final eden.pt. It seems that eden's weights are not coming from corresponding weights in vae and DiT. So I guess for the provided 3 checkpoints, eden.pt != combine(vae's decoder, DiT). Am I missing something?
> 
> Hi, thanks for bringing up this interesting observation!
> 
> After checking our previous code, we found that after **EDEN** was merged, we conducted several thousand steps of end-to-end fine-tuning to further enhance **EDEN**'s capabilities. Therefore, the final weights of `eden.pt` are different from the original `vae.pt` and `dit.pt`.

I checked the code again, there are only train_vae.py and train_dit.py (no end-to-end training script). This is a latent diffusion model, to my understanding, encoder-decoder are usually pretrained first. So how do you perform end-to-end fine-tuning?

**Comment by bbldCVer** [comment]
> > > > Hi, sorry for not providing clear instructions in the README. We've discussed how to obtain eden.pt in [#4](https://github.com/bbldCVer/EDEN/issues/4) — hope that helps!
> > > 
> > > 
> > > Thank you for reply. I manually checked the weights of the pretrained vae, pretrained DiT and the final eden.pt. It seems that eden's weights are not coming from corresponding weights in vae and DiT. So I guess for the provided 3 checkpoints, eden.pt != combine(vae's decoder, DiT). Am I missing something?
> > 
> > 
> > Hi, thanks for bringing up this interesting observation!
> > After checking our previous code, we found that after **EDEN** was merged, we conducted several thousand steps of end-to-end fine-tuning to further enhance **EDEN**'s capabilities. Therefore, the final weights of `eden.pt` are different from the original `vae.pt` and `dit.pt`.
> 
> I checked the code again, there are only train_vae.py and train_dit.py (no end-to-end training script). This is a latent diffusion model, to my understanding, encoder-decoder are usually pretrained first. So how do you perform end-to-end fine-tuning?

Hi, you are correct. To clarify, once you obtain `eden.pt` by merging, you can perform **end-to-end fine-tuning** using a logic similar to `train_dit.py`. The main difference is that the **decoder** is also updated during training. The objective becomes a joint loss: $$\text{Total Loss} = \text{DiT Loss} + \text{VAE Reconstruction Loss}$$
Apologies that the code for this part is currently too messy to be shared officially. However, you can implement this end-to-end fine-tuning yourself by modifying `train_dit.py` based on the logic mentioned above.
